cgp/population.py

```diff
@@ -49,7 +53,10 @@ def __init__(
         self.generation = 0
         self._max_idx = 0  # keeps track of maximal idx in population used to label individuals
 
-        self._generate_random_parent_population()
+        if individual_init is not None and not callable(individual_init):
+            raise TypeError("individual_init must be a callable")
```

Comment:
Sorry, I  have an additional thing to nitpick around: Can this be extended to raise an error (or warning) if `individual_init`is not of type `Callable(IndividualBase) -> IndividualBase`

Reply:
i don't think this is easily possible. the only way to check signatures (that i know of) is via mypy. however, the user would need to execute mypy on their code themselves.